### PR TITLE
Fixed issue #13 to add feature to scroll to top

### PR DESCRIPTION
**Pull request by rayyan016**
devissue
Title: Fix issue #13: Add scroll-to-top button.

Description: 

-Overview: This PR resolves issue #13, which involved adding a scroll-to-top button functionality to the application.

-Changes Made: I created a new ScrollToTop component and integrated it into the existing codebase. The button is now displayed in the bottom-right corner of the page, allowing users to easily scroll back to the top.

### CODE REVIEW

**Comment by ayushgupta4002** [comment]
@rayyan016 looks good to me, can you try improving Design of go to top button?

**Comment by ayushgupta4002** [comment]
@Adi-ty please review

File: client/src/Components/ScrollToTop.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      window.removeEventListener("scroll", toggleVisibility); // Remove event listener when component is unmounted
+    };
+  }, []);
+
+  return (
+    <button
+      className={`scroll-to-top ${isVisible ? "visible" : ""}`}
+      onClick={scrollToTop}
+    >
+      ⬆️
```
Comment (Adi-ty) [inline client/src/Components/ScrollToTop.jsx:39]: Use better icons maybe some kind of library for arrow icon this doesn't look good right now

Comment (rayyan016) [reply client/src/Components/ScrollToTop.jsx:39]: I will start working on it. Thankyou.

**Comment by rayyan016** [comment]
@ayushgupta4002 I have added a font awesome icon for the visual appeal. Please let me know if it is up to the mark. I would be happy to make more changes if required. 
devisssue2

**Comment by ayushgupta4002** [comment]
@rayyan016 
Your Build Checks Are Failing for deployment issue;
it looks fine to me but the build check needs to be passes in order for me to merge it ;
try installing dependencies again and pushing it again ;
image

You can figure out that is a node module error;
image

**Comment by rayyan016** [comment]
> @rayyan016 Your Build Checks Are Failing for deployment issue; it looks fine to me but the build check needs to be passed in order for me to merge it ; try installing dependencies again and pushing it again ; image
> 
> You can figure out that its a node module error;
> 
> image

@ayushgupta4002 I have fixed the error. Please let me know if there is anything else I should do. Thankyou

**Comment by ayushgupta4002** [comment]
looks Good ; Thankyou @rayyan016 !

**Comment by ayushgupta4002** [comment]
@Adi-ty Please review

**Comment by ayushgupta4002** [comment]
@rayyan016 can you please change the background color of button to #0D2F3F

**Comment by rayyan016** [comment]
> @rayyan016 can you please change the background color of button to #0D2F3F

Done.

**Pull request closed without merging**